vms/platformvm/spend.go

```diff
@@ -585,3 +660,32 @@ func produceOutputs(
 		})
 	}
 }
+
+func canBeSpended(utxoState PUTXOState, spendMode spendMode, burn bool) bool {
+	switch spendMode {
+	case spendModeDeposite:
+		return utxoState&PUTXOStateDeposited == 0 && (!burn || utxoState == PUTXOStateTransferable)
+	case spendModeBond:
+		return utxoState&PUTXOStateBonded == 0 && (!burn || utxoState == PUTXOStateTransferable)
+	case spendModeUndeposit:
+		return !burn && utxoState&PUTXOStateDeposited == 1 || burn && utxoState&PUTXOStateBonded == 0
+	case spendModeUnbond:
+		return !burn && utxoState&PUTXOStateBonded == 1 || burn && utxoState&PUTXOStateDeposited == 0
+	}
+	return false
+}
+
+// stateAfterSpending will only work for correct utxoState that can be spended with correct spendMode
+func stateAfterSpending(utxoState PUTXOState, spendMode spendMode) PUTXOState {
+	switch spendMode {
+	case spendModeDeposite:
+		return utxoState & PUTXOStateDeposited
```

Comment:
Shouldn't this be | ?
Maybe I misunderstand here something....

Reply:
oh, you'r right, my bad